### ISSUE TITLE
Only file and data URLs are supported by the default ESM loader

### ISSUE THREAD

**Issue by smeijer** [preactjs/wmr#308]
I'm trying to use the directory import plugin as listed in [the wiki](https://github.com/preactjs/wmr/wiki/Configuration-Recipes#importing-directories-of-files), and it does work in develop mode (`wmr build`).

But when running `wmr build --prerender` it's throwing an resolve error:

```
Error [ERR_UNSUPPORTED_ESM_URL_SCHEME]: Only file and data URLs are supported by the default ESM loader
    at Loader.defaultResolve [as _resolve] (internal/modules/esm/resolve.js:724:11)
    at Loader.resolve (internal/modules/esm/loader.js:97:40)
    at Loader.getModuleJob (internal/modules/esm/loader.js:242:28)
    at ModuleWrap.<anonymous> (internal/modules/esm/module_job.js:50:40)
    at link (internal/modules/esm/module_job.js:49:36)
```

Is this a known limitation? Or is there perhaps something that I can do to fix this?

**Comment by developit** [comment]
Hmm - I believe this is a bug. There should be no scheme in the generated import, since `--prerender` uses production bundles that have all been pre-transformed. Not sure what the root cause of this would be.

**Comment by developit** [comment]
@smeijer just a thought - any chance your `wmr.config.js` was only applying the plugin during development?  It needs to be done using a default-exported function, not the `start` export:

```js
// should work:
export default function (config) {
  config.plugins.push(lsPlugin(config));
}

// won't work (only runs in development):
export function start (config) {
  config.plugins.push(lsPlugin(config));
}
```

**Comment by jorbuedo** [comment]
I'm trying to use the recipe you created in https://github.com/preactjs/wmr/issues/289#issuecomment-752997646

Basically adding a plugin to have absolute imports starting with '~/'
The config is written correctly as a default exported function, not a named one. 
It throws the same error at build.

**Comment by rschristian** [comment]
@jorbuedo That PR is what allows that to become a config that works. As of right now, it will not due to the way user defined plugins are applied after the sort of default plugins.

**Comment by jorbuedo** [comment]
Oh, you mean https://github.com/preactjs/wmr/pull/290
I hadn't seen the PR, just the issue where the code to do the absolute import was shown. 

In any case the `ERR_UNSUPPORTED_ESM_URL_SCHEME` is still an error in other use cases, like the prerender in this twind example: https://github.com/tw-in-js/example-wmr/issues/1

**Comment by rschristian** [comment]
@jorbuedo Sorry, didn't notice you linked to the issue, not the PR.

Have you edited the twind example? I've used it a number of times since Sascha created it, haven't ran into the issue.

**Comment by jorbuedo** [comment]
No edits. Stright from the repo:

<img width="769" alt="example-wmr" src="https://user-images.githubusercontent.com/1725956/106337559-bd430c80-6291-11eb-8af8-9b00ba278ad0.png">

**Comment by thapecroth** [comment]
Hi
I use the example from the same repo. I don't get the error in my local machine.
But I got the same error when deploying to netlify.

**Comment by ludoviclafole** [comment]
I have the exact same problem.
> 'ls:./test' is imported by public/pages/home/index.js, but could not be resolved – treating it as an external dependency
ls is based on a simple copy paste from lsPlugin example.
You can see it there : https://github.com/ludoviclafole/build-with-fun

if i build with prerender option, it fails with : `Error [ERR_UNSUPPORTED_ESM_URL_SCHEME]: Only file and data URLs are supported by the default ESM loader. Received protocol 'ls:'`

If i disable prerender, doesn't crash at build time, but i get : `Access to script at 'ls:./test' from origin 'http://localhost:8080' has been blocked by CORS policy: Cross origin requests are only supported for protocol schemes: http, data, chrome, chrome-extension, chrome-untrusted, https.` (using npm run serve)

I must miss something in the process :/ 
I'll take a look on what makes it crash further to fix it, if it's on wmr side i'll make a PR
But if you have hints on where to find it, i'll take them ;)

**Comment by marvinhagemeister** [comment]
This was resolved with #600. Updating to a recent version of `@wmrjs/directory-import` fixes it 👍
